### PR TITLE
Adding impl/test crumbs for nucDir

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Adding impl/test crumbs for nucDirectory.

## Why is the change being made?

This is part of the on-going requirement work.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [X] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/nucDirectory/tests/test_nuclideBases.py [hunk 1]
```diff
@@ -185,6 +191,12 @@ def test_nucBases_imposeBurnChainTransmutationBulkStatistics(self):
             )  # ternary fission
 
     def test_nucBases_imposeBurn_nuSF(self):
+        """Test the natural abundance values.
```
Comment (albeanth) [inline armi/nucDirectory/tests/test_nuclideBases.py:194]: It doesn't look like this test is related to the natural abundance values. Rather, it tests that the neutrons released per spontaneous fission.
https://github.com/terrapower/armi/blob/98566492147d6f5702233f89d308c279ff7e6353/armi/nucDirectory/nuclideBases.py#L294-L296

File: armi/nucDirectory/nuclideBases.py [hunk 2]
```diff
@@ -15,6 +15,10 @@
 This module provides fundamental nuclide information to be used throughout the framework
 and applications.
 
+.. impl:: Isotopes and isomers can be queried by name, label, MC2-3 ID, MCNP ID, and AAAZZZS ID.
+    :id: I_ARMI_ND_ISOTOPES0
+    :implements: R_ARMI_ND_ISOTOPES
+
```
[the remark this answers is not in the document]
Comment (john-science) [reply armi/nucDirectory/nuclideBases.py:21]: The problem here is that this requirement is about the module-level, global variables:

https://github.com/terrapower/armi/blob/98566492147d6f5702233f89d308c279ff7e6353/armi/nucDirectory/elements.py#L119-L121

But, for VERY good reasons, global stuff doesn't show up in our API docs.

So I have to put the crumb at the module level.

I agree it's weird, but it'll be a few months before I can move this stuff out of the global scope. SO, until then...

[the remark this answers is not in the document]
Comment (albeanth) [reply armi/nucDirectory/nuclideBases.py:21]: ugh. globals. ok.

File: armi/nucDirectory/tests/test_nuclideBases.py (hunk 1 above)
Comment (john-science) [reply armi/nucDirectory/tests/test_nuclideBases.py:194]: Fair, I change the test crumb description.  The important bit is that this test probes data we got from this file:

https://github.com/terrapower/armi/blob/98566492147d6f5702233f89d308c279ff7e6353/armi/nucDirectory/nuclideBases.py#L1139

Thus proving that we stored the data separate from the code.

File: armi/nucDirectory/tests/test_elements.py [hunk 3]
```diff
@@ -35,14 +35,32 @@ def test_elements_elementBulkProperties(self):
...
 
     def test_element_elementBySymbolReturnsElement(self):
+        """Get elements by symbol.
+
+        .. test:: Get elements by symbol.
+            :id: I_ARMI_ND_ELEMENTS2
+            :tests: R_ARMI_ND_ELEMENTS
+        """
         for ee in elements.byZ.values():
             self.assertIs(ee, elements.bySymbol[ee.symbol])
```
[the remark this answers is not in the document]
Comment (john-science) [reply armi/nucDirectory/tests/test_elements.py:65]: Done.

File: armi/nucDirectory/tests/test_nuclideBases.py (hunk 1 above)
Comment (albeanth) [reply armi/nucDirectory/tests/test_nuclideBases.py:194]: Can you also update the docstring for the test since it's wrong?

Comment (john-science) [reply armi/nucDirectory/tests/test_nuclideBases.py]: Done

**Review by albeanth — APPROVED** [review]

**Pull request merged**